tasks.c

```diff
@@ -3684,6 +3684,8 @@ static void prvCheckTasksWaitingTermination( void )
 		pxTaskStatus->pcTaskName = ( const char * ) &( pxTCB->pcTaskName [ 0 ] );
 		pxTaskStatus->uxCurrentPriority = pxTCB->uxPriority;
 		pxTaskStatus->pxStackBase = pxTCB->pxStack;
+		pxTaskStatus->pxTopOfStack = (StackType_t *)pxTCB->pxTopOfStack;

```

Comment:
Please wrap these two lines with checks for configRECORD_STACK_HIGH_ADDRESS==1 and portSTACK_GROWTH>0.

The code must compile when pxTopOfStack and pxEndOfStack are not present in the StackType_t structure.